src/storage/cloud_storage/mod.rs

```diff
@@ -287,37 +283,39 @@ impl<U: Sync + Send> StorageBackend<U> for CloudStorage {
         Ok(Object::new(body.to_vec()))
     }
 
-    async fn put<P: AsRef<Path> + Send, B: tokio::prelude::AsyncRead + Send + 'static>(
+    #[allow(unused)]
+    async fn put<P: AsRef<Path> + Send, B: tokio02::io::AsyncRead + Send + 'static + Unpin>(
         &self,
         _user: &Option<U>,
         bytes: B,
         path: P,
         _start_pos: u64,
     ) -> super::Result<u64> {
-        let uri = self.uris.put(path)?;
-        let token: Token = self.get_token().compat().await?;
-
-        let request: Request<Body> = Request::builder()
-            .uri(uri)
-            .header(header::AUTHORIZATION, format!("{} {}", token.token_type, token.access_token))
-            .header(header::CONTENT_TYPE, APPLICATION_OCTET_STREAM.to_string())
-            .method(Method::POST)
-            .body(Body::wrap_stream(FramedRead::new(bytes, BytesCodec::new()).map(|b| b.freeze())))
```

Comment:
@Nonius this method here doesn't work with tokio 0.2 AsyncRead, perhaps after your change it will.